File: model/modules/NIRv/NIRv_forcing/NIRv_forcing.m
function [f,fe,fx,s,d,p]=NIRv_forcing(f,fe,fx,s,d,p,info,tix)
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
% sets the value of s.cd.NIRv from the forcing in every time step
%
% Inputs:
%   - f.NIRv read from the forcing data set
%
% Outputs:
%   - s.cd.NIRv: the value of NIRv for current time step
%
% Modifies:
%   - s.cd.NIRv
%
% References:
%   - 
%
% Created by:
%   - Simon Besnard (sbesnard)
%
% Versions:
%   - 1.0 on 29.04.2020 (sbesnard): 
%
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

%% 
s.cd.NIRv = f.NIRv(:,tix);

end

File: src/Models/NIRv/NIRv_forcing.jl
export NIRv_forcing

struct NIRv_forcing <: NIRv end

function compute(params::NIRv_forcing, forcing, land, helpers)
    ## unpack forcing
    @unpack_nt f_NIRv ⇐ forcing

    NIRv = f_NIRv
    
    ## pack land variables
    @pack_nt NIRv ⇒ land.states
    return land
end

purpose(::Type{NIRv_forcing}) = "sets the value of land.states.NIRv from the forcing in every time step"

@doc """

$(getBaseDocString())

---

# Extended help

*References*

*Versions*
 - 1.0 on 29.04.2020 [sbesnard]

*Created by:*
 - sbesnard
"""
NIRv_forcing
